railties/lib/rails/generators/erb/scaffold/templates/_form.html.erb.tt

```diff
@@ -24,6 +24,9 @@
 <% elsif attribute.attachments? -%>
     <%%= form.label :<%= attribute.column_name %> %>
     <%%= form.<%= attribute.field_type %> :<%= attribute.column_name %>, multiple: true %>
+<% elsif %i(float decimal).include?(attribute.type) -%>
+    <%%= form.label :<%= attribute.column_name %> %>
+    <%%= form.<%= attribute.field_type %> :<%= attribute.column_name %>, step: :any %>
```

Comment:
Can we add a method to return tag attributes as well as and use it?
The only difference between this code and the existing code is the specification of attributes, so if we can share it, we can avoid the complexity of a template file.

Reply:
no, because the :step attribute... It should be :any for float/decimal to provide decimal separator.

Reply:
How about the following? 

```diff 
diff --git a/railties/lib/rails/generators/erb/scaffold/templates/_form.html.erb.tt b/railties/lib/rails/generators/erb/scaffold/templates/_form.html.erb.tt
index 1dddc3d698..0b5ac63308 100644
--- a/railties/lib/rails/generators/erb/scaffold/templates/_form.html.erb.tt
+++ b/railties/lib/rails/generators/erb/scaffold/templates/_form.html.erb.tt
@@ -21,12 +21,9 @@
   <div class="field">
     <%%= form.label :password_confirmation %>
     <%%= form.password_field :password_confirmation %>
-<% elsif attribute.attachments? -%>
-    <%%= form.label :<%= attribute.column_name %> %>
-    <%%= form.<%= attribute.field_type %> :<%= attribute.column_name %>, multiple: true %>
 <% else -%>
     <%%= form.label :<%= attribute.column_name %> %>
-    <%%= form.<%= attribute.field_type %> :<%= attribute.column_name %> %>
+    <%%= form.<%= attribute.field_type %> :<%= attribute.column_name_with_tag_option %> %>
 <% end -%>
   </div>
 
diff --git a/railties/lib/rails/generators/generated_attribute.rb b/railties/lib/rails/generators/generated_attribute.rb
index 377a5dfc65..3ae8bbee1b 100644
--- a/railties/lib/rails/generators/generated_attribute.rb
+++ b/railties/lib/rails/generators/generated_attribute.rb
@@ -74,20 +74,30 @@ def initialize(name, type = nil, index_type = false, attr_options = {})
 
         end
       end
 
+      def form_tag_option
+        case type
+        when :float, :decimal then "step: :any"
+        when :attachments     then "multipe: :true"
+        end
+      end
+
+      def column_name_with_tag_option
+        [column_name, form_tag_option].compact.join(", ")
+      end
```

Reply:
Its ok...

Reply:
@lazaronixon how do you feel about accepting this suggestion so that the PR can move forward?

Reply:
I 100% agree

Reply:
merged this changes!!

Reply:
Reviewing better I noticed this change introduce errors and reverted to my original change.